tests/shell/tests.py

```diff
@@ -29,3 +39,25 @@ def test_stdin_read(self, select):
             stdin.seek(0)
             call_command('shell')
         self.assertEqual(stdout.getvalue().strip(), '100')
+
+    # If we don't patch select then the following tests will fail when the test
+    # suite is run in parallel mode.  This is because the the tests are run in
+    # a subprocess and the subprocess's stdin is closed and replaced by
+    # /dev/null.  Reading from /dev/null always returns EOF and so select will
+    # always show that sys.stdin is ready to read.  This causes problems
+    # because of the call to select.select towards the end of
+    # django.core.management.commands.shell.BaseCommand.handle.
+
+    @unittest.skipIf(IPython is not None, 'IPython is installed')
+    @mock.patch('django.core.management.commands.shell.select.select')
+    def test_shell_with_ipython_not_installed(self, select):
+        select.return_value = [[], [], []]
+        with self.assertRaisesMessage(CommandError, "Couldn't import ipython interface."):
+            call_command('shell', interface='ipython')
+
+    @unittest.skipIf(bpython is not None, 'bython is installed')
```

Comment:
not entirely sure of the value of both tests since it's the same code path they're testing now, but they're fast and can't hurt right?